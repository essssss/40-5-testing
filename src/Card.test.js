import { render, fireEvent } from "@testing-library/react";
import Card from "./Card";
import TEST_IMAGES from "./_testCommon.js";

const cardImage = TEST_IMAGES[0];
it("renders without crashing", () => {
    render(
        <Card
            caption="test-caption"
            src={cardImage.src}
            currNum="1"
            totalNum={TEST_IMAGES.length}
        />
    );
});
it("should match snapshot", () => {
    const { asFragment } = render(
        <Card
            caption="test-caption"
            src={cardImage.src}
            currNum="1"
            totalNum={TEST_IMAGES.length}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});
