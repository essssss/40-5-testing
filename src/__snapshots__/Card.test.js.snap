// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      test-caption
    </h4>
    <img
      alt="test-caption"
      class="Card-image"
      src="test1.com"
    />
    <small
      class="Card-small"
    >
      Image 1 of 3.
    </small>
  </div>
</DocumentFragment>
`;
